src/table/hooks/__tests__/use-snapshot.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { generateLayout } from '../../../__test__/generate-test-data';
+import { TableLayout, ViewService } from '../../../types';
+import { getVisibleHeight } from '../use-snapshot';
+
+describe('use-snapshot', () => {
```

Comment:
You also need to test the early return, when visibleRowEndIndex < 0

Reply:
test case added.